File: RuboCop/Ruby 2/Ruby2.cpp
#include <fstream>
#include <iostream>
#include <iomanip>
using namespace std;
int main()
{
    int n, sk, nsk=0, dsk, sk2, apverstas=0;
    cout<<"iveskite norima skaiciu:"<<endl;
    cin>>n;


    do
    {
        sk = n % 10;
        dsk=10*sk+sk;
        nsk=(100*nsk)+dsk;

        n = n / 10;
    }
    while (n != 0);

    do
    {
        sk2=nsk%10;
        apverstas=apverstas*10+sk2;
        nsk/=10;
    }
    while(nsk!=0);
    cout<<"padvigubintas skaicius="<<apverstas<<endl;
    return 0;
}

File: app/RuboCop/Ruby 2/ruby2.rb
# frozen_string_literal: true

# Ivedus skaiciu, dublikuoti kiekviena jo skaiciu, pvz.:
# ivedus 123, turetu gautis 112233, 112 - 111122 ir t.t.
puts 'iveskite norima skaiciu'
n = gets.chomp

while n != 0
  sk = n.to_i % 10
  dsk = 10 * sk.to_i + sk.to_i
  nsk = (100 * nsk.to_i) + dsk.to_i
  n = n.to_i / 10
end
while nsk != 0
  sk2 = nsk.to_i % 10
  apverstas = apverstas.to_i * 10 + sk2.to_i
  nsk = nsk.to_i / 10
end
puts 'padvigubintas skaicius: ' + apverstas.to_s
